controllers/promise_controller_test.go

```diff
@@ -52,17 +52,17 @@ var _ = Describe("PromiseController", func() {
 			})
 		})
 
-		When("the promise is created via a PromiseRelease", func() {
+		When("the promise contains a version label", func() {
 			BeforeEach(func() {
-				promise.Labels["kratix.io/promise-release-version"] = "v1.2.3"
+				promise.Labels["kratix.io/promise-version"] = "v1.2.3"
 				Expect(fakeK8sClient.Create(ctx, promise)).To(Succeed())
 
 				result, err := reconcile(&reconciler, promise, &opts{singleReconcile: true})
 				Expect(err).NotTo(HaveOccurred())
 				Expect(result).To(Equal(ctrl.Result{}))
 			})
 
-			It("persists the value labels in the Status", func() {
+			It("it gets its status updated with the version", func() {
```

Comment:
why is this the behaviour 🤔 shouldn't we just set the status directly when we create the promise in the promiserelease controller

Reply:
afaik you need to do 2 client calls to set status in the promise controller?

Reply:
yeah, but it feels odd that the label is essentially an override mechanism

Reply:
or is that the behaviour to unluck manual version setting?